vllm/distributed/parallel_state.py

```diff
@@ -125,6 +129,7 @@ class GroupCoordinator:
     pynccl_comm: Optional[Any]  # PyNccl communicator
     ca_comm: Optional[Any]  # Custom allreduce communicator
     mq_broadcaster: Optional[Any]  # shared memory broadcaster
+    use_xla: bool  # Whether to use PyTorch XLA communicator
```

Comment:
`use_xxx` is a initialization parameter, and we usually hold `communicator` inside group coordinator.

Can you add a `tpu_communicator` under https://github.com/vllm-project/vllm/tree/main/vllm/distributed/device_communicators ?

One additional benefit, is that you can implement the `gather` logic to `allgather`, without intrusive change to `logits_processor.py` .

Reply:
> One additional benefit, is that you can implement the gather logic to allgather, without intrusive change to logits_processor.py .

This is actually not the case because the TPU backend explicitly requires all-gather, which means each device's output should not be None. If we implement gather by using `all-gather` and outputting None for non-root ranks, XLA will raise an error.